### ISSUE TITLE
label error rate or label correct rate ?

### ISSUE THREAD

**Issue by anxingle** [igormq/ctc_tensorflow_example#3]
I see this comment   [https://github.com/igormq/ctc_tensorflow_example/blob/master/ctc_tensorflow_example.py#L145](https://github.com/igormq/ctc_tensorflow_example/blob/master/ctc_tensorflow_example.py#L145).       
I wonder  tf.reduce_mean(tf.edit_distance(tf.cast(decoded[0], tf.int32),targets)) should be correct rate,so it just be clerical error or something else?  

By the way,the [code](https://gist.github.com/igormq/eff5b2196a52e89c61ea52515ed87c47) you give me works very well on SVHN dataset  when I  added Dropout and  one more( now is 2 layers) layer  .  Amazing!

**Comment by anxingle** [comment]
Got it!     
 Cost will be more an more smaller,and acc is also smaller,but  [train_ler](https://github.com/igormq/ctc_tensorflow_example/blob/master/ctc_tensorflow_example.py#L166) becomes more and more bigger.

**Comment by igormq** [comment]
@anxingle this label error rate (LER) is the [Levenshtein distance](https://en.wikipedia.org/wiki/Levenshtein_distance) and was used by Graves in his thesis. The `acc` (sorry for this name, this is not the accuracy in the strict mean, I'll change this), who is responsible for the calculation of LER,  should decrease (in mean) after each epoch. So, `train_ler` should decrease too, because this is an accumulator of `acc` to get the mean across each epoch.

**Comment by anxingle** [comment]
The [gist](https://gist.github.com/igormq/eff5b2196a52e89c61ea52515ed87c47) finally  print "Test label error rate:  0.93" . So about 93% of  prediction fail to predict successfully ?

**Comment by igormq** [comment]
Yes, that's right.

**Comment by XuesongYang** [comment]
I wonder if tf.edit_distance is correctly used in this code snippet. Levenshtein distance measures minimum number of edits for characters from one sequence (hyp) to another (ref). Therefore, the output score normalized by the length of ref could be greater than 1, e.g. hyp='abcdef', ref='a', Levenshtein distance= 5 / 1 = 5. However, the evaluation on ASR performance is slightly different, since word/phone error rate is constrained in the range of [0, 1]. 

Please correct me if I get wrong understanding. Thanks.

**Comment by igormq** [comment]
Hi @XuesongYang , it is not constrained in the range of [0, 1], as you can see [here](ftp://ftp.idsia.ch/pub/juergen/icml2006.pdf) in the original Alex Graves' paper, the LER is the Edit distance normalized by the sequence length.
